lib/improver/blending/blend_across_adjacent_points.py

```diff
@@ -119,52 +127,50 @@ def correct_collapsed_coordinates(orig_cube, new_cube, coords_to_correct):
+                Cube containing input for blending.
 
         Returns:
-            cube (iris.cube.Cube):
+            blended_cube (iris.cube.Cube):
                 The processed cube, with the same coordinates as the input
-                cube. The points in one coordinate will be blended with the
-                adjacent points based on a triangular weighting function of the
-                specified width.
+                central_cube. The points in one coordinate will be blended
+                with the adjacent points based on a triangular weighting
+                function of the specified width.
 
         """
         # We need to correct all the coordinates associated with the dimension
         # we are collapsing over, so find the relevant coordinates now.
         dimension_to_collapse = cube.coord_dims(self.coord)
         coords_to_correct = cube.coords(dimensions=dimension_to_collapse)
         coords_to_correct = [coord.name() for coord in coords_to_correct]
-        # We will also need to correct the bounds on these coordinates,
-        # as bounds will be added when the blending happens, so add bounds if
-        # it doesn't have some already.
-        for coord in coords_to_correct:
-            if not cube.coord(coord).has_bounds():
-                cube.coord(coord).guess_bounds()
         # Set up a plugin to calculate the triangular weights.
         WeightsPlugin = ChooseDefaultWeightsTriangular(
             self.width, units=self.parameter_units)
         # Set up the blending function, based on whether weighted blending or
         # maximum probabilities are needed.
         BlendingPlugin = WeightedBlendAcrossWholeDimension(self.coord,
                                                            self.mode)
-        result = iris.cube.CubeList([])
-        # Loop over each point in the coordinate we are blending over, and
-        # calculate a new weighted average for it.
-        for cube_slice in cube.slices_over(self.coord):
-            point = cube_slice.coord(self.coord).points[0]
-            weights = WeightsPlugin.process(cube, self.coord, point)
-            blended_cube = BlendingPlugin.process(cube, weights)
-            self.correct_collapsed_coordinates(cube_slice, blended_cube,
-                                               coords_to_correct)
-            result.append(blended_cube)
-        result = concatenate_cubes(result)
-        return result
+        # Convert central point into the units of the cube, so that a
+        # central point can be extracted.
+        self.central_point = (
+            Unit(self.parameter_units).convert(
+                self.central_point, cube.coord(self.coord).units))
+        constr = iris.Constraint(
+            coord_values={self.coord: self.central_point})
+        central_point_cube = cube.extract(constr)
+        # Check that the central point is allowed.
+        check_point_within_allowed_range(cube, self.coord, self.central_point)
```

Comment:
You've already extracted the central point, so is this enough, without having to check the point is within the range of the cube. If you want to keep this function, you probably want to call it before extracting the point from the cube.